Automattic/jetpack
# Protect: Add support for displaying file and database threats

This PR adds the display of file and database threats to the Protect UI.

#### Changes proposed in this Pull Request:

* Adds new `<MarkedLines />` and `<DiffViewer />` components, derived from their existing versions in Calypso.
* Installs the `diff` package, a dependency of the `<DiffViewer />` component. This triggers an update to the root pnpm lock file, which will need to be synced with WPCOM.
* Adds `files` and `database` properties to `useProtectData()` to make these threats accessible to various components.
* Integrates the new `files` and `database` properties, alongside the new `<MarkedLines />` and `<DiffViewer />` components, to the existing threat list navigation and results.

#### Other information:


#### Jetpack product discussion



1201069996155224-as-1202738501526869

#### Does this pull request change what data or activity we track or use?



Yes - includes new `jetpack_protect_navigation_file_click` and `jetpack_protect_navigation_database_click` tracks events [here](https://github.com/Automattic/jetpack/pull/26366/files#diff-2439f5bc4c4dba3347f4251f222178ce011a81080147220e58cf66a26b811ffcR37-R43).

#### Testing instructions:






* Add vulnerabilities to your site with the Jetpack Debug Helper - it is recommended to test both with the Scan and Protect helpers, the Scan helper will add actual test vulnerabilities to your site while the Protect helper will mock the results of the Scan API.
* Validate that file and database threats do not appear when on a free plan. The navigation should not include the "Files" or "Database" items.
* Upgrade your site to a plan that includes Jetpack scan.
* Validate that the scan results displayed for your site in Calypso match the results displayed by Protect.

#### Screenshots

<img width="1120" alt="Screen Shot 2022-09-24 at 12 46 01 PM" src="https://user-images.githubusercontent.com/10933065/192114090-5b94a8b4-30ad-43a2-b801-32bc5411bad4.png">
